README.md

```diff
@@ -12,10 +14,92 @@ npm install --save "git://github.com/ditrit/leto-modelizer-plugin-core.git"
 
 ### Usage
 
+Here is all import you can use in your plugin :
+
+```js
+import {
+  Component,
+  ComponentDefinition,
+  ComponentAttributeDefinition,
+  ComponentDrawOption,
+  ComponentAttribute,
+  ComponentLink,
+  DefaultDrawer,
+  DefaultMetadata,
+  DefaultParser,
+  DefaultRender,
+  ParseError
+} from "leto-modelizer-plugin-core";
+```
+
+For more information about all imports please refer to technical documentation and project template.
+
+You can use it in that way:
+
 ```js
-import { renderString, LetoObjectNode } from "leto-modelizer-plugin-core";
+import { DefaultParser } from 'leto-modelizer-plugin-core';
+
+class FruitParser extends DefaultParser {
+  parse(inputs) {
+    // Write your own parser here
+    return {
+      components: [/* ... */], // Generated components from parse
+      links: [/* ... */]       // Generated component links from parse
+    }
+  }
+}
+
+export default FruitParser;
 ```
 
+To see complete example, please refer to [the fruit example](https://github.com/ditrit/leto-modelizer-plugin-template/tree/main/examples/fruit-plugin) in project template.
+
+## How to create your plugin
+
+You can use [leto-modelizer-plugin-template](https://github.com/ditrit/leto-modelizer-plugin-template) project as template to create your own plugin.
+
+The project template leto-modelizer-plugin-template provides you the default structure of a plugin and all useful scripts to generate resources and other.
+
+It is strongly recommended to use it and all next explanations will make reference to it.
+
+Furthermore, in this template there are code comments to explain how to override methods/classes and usages. 
+
+### Default Plugin structure
+
+For your plugin to be used by `Leto Modelizer`, it needs to have this structure:
+
+```js
+// src/index.js
+export default {
+  PluginDrawer: MyPluginDrawer,     // MyPluginDrawer has to extend DefaultDrawer
+  PluginMetadata: MyPluginMetadata, // MyPluginMetadata has to extend DefaultMetadata
+  PluginParser: MyPluginParser,     // MyPluginParser has to extend DefaultParser
+  PluginRenderer: MyPluginRenderer, // MyPluginRenderer has to extend DefaultRender
+  resources,
+};
+```
+
+`resources` represents an object that contains all your assets for the plugin.
+By default, the project template provide a `DefaultModel.svg` and a `DefaultIcon.svg`.
+
+The object `resources` is managed by a script in project template for its generation.
+
+### How is it working
+
+| Plugin lifecycle |
+| :--: |
+| <img height="400" src="/documentations/lifecycle.svg"/> |
+
+This is default lifecycle of plugin usage in Leto Modelizer.
+
+### Plugin creation
+
+Create you plugin project from [leto-modelizer-plugin-template](https://github.com/ditrit/leto-modelizer-plugin-template) and follow the readme section of the template project.
+
+### Technical documentation
+
+Technical documentation can be found [here](https://ditrit.io/leto-modelizer-plugin-core/).
```

Comment:
Maybe put this link on a larger word or group of word

Reply:
no it's ok :)